File: cs710flutterapp/lib/models/rfid_inventory_stats.dart
/// Statistics for RFID inventory operations
class RfidInventoryStats {
  final int uniqueTagCount;
  final int totalReads;
  final double readRate; // tags per second
  final int elapsedTimeMs;

  const RfidInventoryStats({
    required this.uniqueTagCount,
    required this.totalReads,
    required this.readRate,
    required this.elapsedTimeMs,
  });

  /// Create RfidInventoryStats from platform channel map
  factory RfidInventoryStats.fromMap(Map<String, dynamic> map) {
    return RfidInventoryStats(
      uniqueTagCount: map['uniqueTagCount'] as int,
      totalReads: map['totalReads'] as int,
      readRate: (map['readRate'] as num).toDouble(),
      elapsedTimeMs: map['elapsedTimeMs'] as int,
    );
  }

  /// Convert to map for platform channel
  Map<String, dynamic> toMap() {
    return {
      'uniqueTagCount': uniqueTagCount,
      'totalReads': totalReads,
      'readRate': readRate,
      'elapsedTimeMs': elapsedTimeMs,
    };
  }

  /// Get elapsed time in seconds
  int get elapsedSeconds => (elapsedTimeMs / 1000).floor();

  @override
  String toString() =>
      'RfidInventoryStats(unique: $uniqueTagCount, total: $totalReads, rate: $readRate tags/s)';
}

File: csl-rfid-android-sdk/src/main/java/com/csl/rfidsdk/models/RfidInventoryStats.java
package com.csl.rfidsdk.models;

/**
 * Statistics for RFID inventory operations
 */
public class RfidInventoryStats {
    private final int uniqueTagCount;
    private final int totalReads;
    private final long duration;
    private final double readRate;
    private final long startTime;

    public RfidInventoryStats(int uniqueTagCount, int totalReads, long duration, double readRate, long startTime) {
        this.uniqueTagCount = uniqueTagCount;
        this.totalReads = totalReads;
        this.duration = duration;
        this.readRate = readRate;
        this.startTime = startTime;
    }

    public int getUniqueTagCount() {
        return uniqueTagCount;
    }

    public int getTotalReads() {
        return totalReads;
    }

    public long getDuration() {
        return duration;
    }

    public double getReadRate() {
        return readRate;
    }

    public long getStartTime() {
        return startTime;
    }

    @Override
    public String toString() {
        return "RfidInventoryStats{" +
                "uniqueTagCount=" + uniqueTagCount +
                ", totalReads=" + totalReads +
                ", duration=" + duration +
                ", readRate=" + readRate +
                '}';
    }

    /**
     * Builder for RfidInventoryStats
     */
    public static class Builder {
        private int uniqueTagCount;
        private int totalReads;
        private long elapsedTimeMs;
        private double readsPerSecond;

        public Builder uniqueTagCount(int uniqueTagCount) {
            this.uniqueTagCount = uniqueTagCount;
            return this;
        }

        public Builder totalReads(int totalReads) {
            this.totalReads = totalReads;
            return this;
        }

        public Builder elapsedTimeMs(long elapsedTimeMs) {
            this.elapsedTimeMs = elapsedTimeMs;
            return this;
        }

        public Builder readsPerSecond(double readsPerSecond) {
            this.readsPerSecond = readsPerSecond;
            return this;
        }

        public RfidInventoryStats build() {
            return new RfidInventoryStats(uniqueTagCount, totalReads, elapsedTimeMs, readsPerSecond, System.currentTimeMillis());
        }
    }
}
